rfcs/078-name-reconciliation-service/README.md

```diff
@@ -0,0 +1,330 @@
+- LLM Adjudicator: ChatSession abstraction over Bedrock ChatBedrock client (temperature=0) with LangGraph memory (currently unused beyond session scoping).
+- Output Artifacts: Per-target JSON containing:
+
+  ```json
+  {
+    "label": "<target label>",
+    "idx": "<WellcomeId>_<FaissIndex>",
+    "reconciled_labels": [ { "label": "<matched label>", "idx": "<WellcomeId>_<FaissIndex>" } ],
+    "candidates": [ { "label": "<candidate label>", "idx": "<WellcomeId>_<FaissIndex>", "similarity": <float> } ]
+  }
+  ```
+
+#### NARESE Per Sample Sequence Diagram
+
+![narese_sequence.png](narese_sequence.png)
+
+- Architecture Diagrams: `docs/name_rec_architecture.puml` (component & data), `docs/name_rec_sequence.puml` (per-record sequence).
+
+### Current PoC vs Target Production
+
+| Aspect | Current PoC | Target Production |
+|--------|-------------|-------------------|
+| Input Source | Static CSV extracted from *denormalised Elasticsearch works index* (snapshot taken July 2025) -> `dn_labels_dedup_data.csv` | Live incremental feed (stream or periodic export) from canonical Works & Entities stores |
+| Execution Mode | Manual batch script (`name_rec_run.py`) sampling ~3000 rows for evaluation | On-demand API + scheduled full refresh job |
+| Storage | Per-record JSON files on local filesystem | Durable object storage + backing relational / graph store for clusters |
+| Index Refresh | Manual, ad-hoc FAISS build | Automated pipeline with dimension / checksum guard & blue/green swap |
+| Reconciliation Engine | Embedding + Bedrock LLM per target | Same core flow with caching, possible lightweight rules model for high-confidence auto matches |
+| Monitoring | None | Metrics (latency, success rate, empty cluster ratio) + structured logs + tracing |
+| Governance | Developer-only | Product + Data curation review workflow (UI) |
+
+The current PoC validates technical feasibility (semantic retrieval + LLM filtering) using a static July 2025 snapshot. This RFC defines the *production API* that will sit in front of an evolved pipeline able to serve real-time reconciliation queries and batch enrichment.
+
+### Reconciliation Logic (LLM Rule Set)
+
+System prompt enforces:
+
+- Require medium disambiguation level: full name match OR initials + supporting date.
+- Initial-based matches without date are low certainty and excluded unless promoted by additional context.
+- Ambiguity => return empty list (preference for precision over recall initially).
+- Output strict JSON list of `id_idx` strings.
+
+### File / Data Contracts
+
+| Artifact | Purpose | Frequency | Notes |
+|----------|---------|-----------|-------|
+| `dn_labels_dedup_data.csv` | Source label universe (name retains historical 'dedup' misnomer) | Input snapshot | Legacy naming from earlier Dedupe lib prep; NOT a strictly deduplicated set |
+| `name_rec_faiss.index` | Embedding ANN search | Rebuilt on data refresh | Must align embedding dimension |
```

Comment:
It would be interesting to know how big the FAISS index is, as that potentially impacts future infrastructure provisioning.

Reply:
1.6gb seems very manageable!